### PR TITLE
Sample upload

### PR DESCRIPTION
**Pull request by zelzhan**
**Key features:**
- Added frontend and backend libraries for the sample upload. 
- User uploads the sample to the sample platform, it accepts it and saves it to the bucket. 
- As an additional feature, now libraries can be imported much easier: instead of typing: `../../../../../../` every time, now the libraries can be accessed as follows : `@new-sample-platform/api-implementation/library` 

ezgif com-video-to-gif

The progress bar is filling immediately because I'm hosting locally and the speed of transfer is enormous

### CODE REVIEW

File: apps/api/src/app/app.module.ts [hunk 1]
```diff
@@ -2,11 +2,17 @@ import { Module } from '@nestjs/common';
 
 import { AppController } from './app.controller';
 import { VMOrchestrationModule } from '../../../../libs/api-implementation/orchestration/src';
-import { ApiImplementationRegressionTestsModule } from '../../../../libs/api-implementation/regression-tests/src'
-import { ApiImplementationDatabaseModule } from '../../../../libs/api-implementation/database/src'
+import { ApiImplementationRegressionTestsModule } from '@new-sample-platform/api-implementation/regression-tests';
```
Comment (canihavesomecoffee) [inline apps/api/src/app/app.module.ts:5]: This looks nice, but can you change it so it drops the "new" part? Just "sample-platform" should be sufficient, and "new" will be outdated at some point in the future ;)

**Review by canihavesomecoffee — APPROVED** [review]

File: apps/api/src/app/app.module.ts (hunk 1 above)
Comment (zelzhan) [reply apps/api/src/app/app.module.ts:5]: Yeap, it is doable pretty easily. It depends on the name of the repository. Right now I have an application called `sample-platform` so I guess that will cause some conflict. I think I should change that to the `frontend` or so, and change the name of the repo to `sample-platform`. But I guess that's not the highest priority task, can I postpone it till the more important tasks will be done?

Comment (canihavesomecoffee) [reply apps/api/src/app/app.module.ts:5]: If it's not that difficult to achieve, no problem waiting until other more important things are done.

Comment (thealphadollar) [reply apps/api/src/app/app.module.ts:5]: Yeah +1 to both of you.

Comment (thealphadollar) [reply apps/api/src/app/app.module.ts:5]: A reminder I've created, https://github.com/zelzhan/new-sample-platform/issues/14

File: libs/api-implementation/samples/src/lib/samples.controller.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@Controller('samples')
+export class SamplesController {
+  constructor(private service: SamplesService) {}
+
+  @Post('upload')
+  @UseInterceptors(FileInterceptor('file'))
+  uploadFile(@UploadedFile() file) {
+    this.service.writeToBucket(file);
+    return 200 
+  }
```
Comment (thealphadollar) [inline libs/api-implementation/samples/src/lib/samples.controller.ts]: I'm curious to understand how are we handling errors here? @zelzhan Can you please help me?

**Review by thealphadollar — APPROVED** [review]
I'm so glad to see this much progress, I'm pretty sure and have trust that you'd be able to deliver more than we thought during the second evaluations @zelzhan . Thanks a lot for working so hard <3

I've one small doubt that I've listed in comments. Apart from that, I want to know are we checking for file types that are allowed to be uploaded? Are we planning to have some whitelisting and blacklisting?

Also, I'm really poor at reviewing front-end code; I tend to not understand ti well. Therefore, I would be depending on you and the GIF and screenshots that you add in the description. Thanks for including a really awesome one covering everything from start to end this time.

File: libs/api-implementation/samples/src/lib/samples.controller.ts (hunk 2 above)
Comment (zelzhan) [reply libs/api-implementation/samples/src/lib/samples.controller.ts]: Thank you for pointing out. Actually I've missed error handling here :). I've updated PR.

Comment (zelzhan) [reply libs/api-implementation/samples/src/lib/samples.controller.ts]: Regarding the blacklisting and whitelisting: right now we are not checking the type of the upload. But yes, this is quite crucial. I'll create issues with the TODOs soon.

Comment (thealphadollar) [reply libs/api-implementation/samples/src/lib/samples.controller.ts]: Sounds good; thank you for clearing my doubts and taking my suggestions :D

File: libs/api-implementation/samples/src/lib/samples.controller.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+export class SamplesController {
+  constructor(private service: SamplesService) {}
+
+  @Post('upload')
+  @UseInterceptors(FileInterceptor('file'))
+  uploadFile(@UploadedFile() file) {
+    try {
+      this.service.writeToBucket(file);
+    } catch (error) {
+      return 'Error while writing the file to the bucket'
```
Comment (thealphadollar) [inline libs/api-implementation/samples/src/lib/samples.controller.ts:20]: Does this handle sending appropriate error code as well?

File: libs/api-implementation/samples/src/lib/services/samples.service.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    try {
+      writeFileSync(
+        AppConfig.PATH_TO_MOUNTED_BUCKET + file.originalname,
+        file.buffer
+      );
+    } catch (error) {
+      console.error(`Error occured while writing the file to the bucket`);
+      console.debug(error.stack);
+      throw error;
+    }
```
Comment (thealphadollar) [inline libs/api-implementation/samples/src/lib/services/samples.service.ts:16]: Amazing :sparkles:

**Review by thealphadollar — APPROVED** [review]

File: libs/api-implementation/samples/src/lib/samples.controller.ts (hunk 3 above)
Comment (thealphadollar) [reply libs/api-implementation/samples/src/lib/samples.controller.ts:20]: @zelzhan Can you please address my doubt here?

**Pull request merged**